### PR TITLE
scheduler: add operatorLimitCounter metrics for each scheduler

### PR DESCRIPTION
**Pull request by Yisaer**
Signed-off-by: Song Gao <disxiaofei@163.com>

### What problem does this PR solve?

close https://github.com/tikv/pd/issues/3350

### What is changed and how it works?

Add metrics to record operator creating failed due to limit configuration.

### Release note
* No release note

### CODE REVIEW

File: server/schedulers/hot_region.go [hunk 1]
```diff
@@ -162,12 +162,23 @@ func (h *hotScheduler) IsScheduleAllowed(cluster opt.Cluster) bool {
 }
 
 func (h *hotScheduler) allowBalanceLeader(cluster opt.Cluster) bool {
-	return h.OpController.OperatorCount(operator.OpHotRegion) < cluster.GetOpts().GetHotRegionScheduleLimit() &&
-		h.OpController.OperatorCount(operator.OpLeader) < cluster.GetOpts().GetLeaderScheduleLimit()
+	hotRegionAllowed := h.OpController.OperatorCount(operator.OpHotRegion) < cluster.GetOpts().GetHotRegionScheduleLimit()
+	leaderAllowed := h.OpController.OperatorCount(operator.OpLeader) < cluster.GetOpts().GetLeaderScheduleLimit()
+	if hotRegionAllowed {
```
Comment (disksing) [inline server/schedulers/hot_region.go:167]: !?

Comment (Yisaer) [reply server/schedulers/hot_region.go:167]: good catch.

**Review by disksing — APPROVED** [review]

**Review by lhy1024 — APPROVED** [review]

File: plugin/scheduler_example/evict_leader.go [hunk 2]
```diff
@@ -205,7 +205,11 @@ func (s *evictLeaderScheduler) Cleanup(cluster opt.Cluster) {
 }
 
 func (s *evictLeaderScheduler) IsScheduleAllowed(cluster opt.Cluster) bool {
-	return s.OpController.OperatorCount(operator.OpLeader) < cluster.GetOpts().GetLeaderScheduleLimit()
+	allowed := s.OpController.OperatorCount(operator.OpLeader) < cluster.GetOpts().GetLeaderScheduleLimit()
+	if !allowed {
+		operator.OperatorLimitCounter.WithLabelValues(s.GetType(), operator.OpLeader.String()).Inc()
```
Comment (HunDunDM) [inline plugin/scheduler_example/evict_leader.go:210]: Maybe statistics can be done where IsScheduleAllowed is called?

File: server/schedulers/balance_leader.go [hunk 3]
```diff
@@ -127,7 +127,11 @@ func (l *balanceLeaderScheduler) EncodeConfig() ([]byte, error) {
 }
 
 func (l *balanceLeaderScheduler) IsScheduleAllowed(cluster opt.Cluster) bool {
-	return l.opController.OperatorCount(operator.OpLeader) < cluster.GetOpts().GetLeaderScheduleLimit()
+	allowed := l.opController.OperatorCount(operator.OpLeader) < cluster.GetOpts().GetLeaderScheduleLimit()
+	if !allowed {
+		operator.OperatorLimitCounter.WithLabelValues(l.GetType(), operator.OpLeader.String()).Inc()
```
Comment (HunDunDM) [inline server/schedulers/balance_leader.go:132]: ditto

**Pull request merged**